### ISSUE TITLE
What effective batch size was used for training?

### ISSUE THREAD

**Issue by yohan-pg** [OpenRobotLab/AnySplat#96]
Hi, 

If I understand the code correctly, the `batch_size: 4` config gets ignored and instead a batch size is determined dynamically. On DL3DV with an H100 (80Gb memory), this appears to always result in a batch size of 1. 

Am I correct in understanding that, when training on 16 A800 cards, the effective batch size ends up being 16?

Thank you!
Yohan

**Comment by 20713** [comment]
> 你好，
> 
> 如果我理解代码没错，`batch_size: 4`配置会被忽略，批处理大小会动态确定。在配备 H100（80GB 内存）的 DL3DV 上，批处理大小似乎总是为 1。
> 
> 我的理解是否正确，当使用 16 张 A800 卡进行训练时，有效批次大小最终为 16？
> 
> 谢谢！ 尤汉
Hi, i have the same question. In the generated config after launch, batch_size is set to 1, but during training the actual batch size seems to be determined dynamically (see the logs below).
B: 1 V: 15 H: 448 W: 448
total_loss: 0.4352969229221344
scene scale: 1.181, pixel-wise num: 3763200, after voxelize: 2462204, voxelize ratio: 0.654
Gaussians attributes:
opacities: mean: 0.3069624900817871, min: 0.16451668739318848, max: 0.5760654807090759
scales: mean: 0.0015245289541780949, min: 0.0006661772495135665, max: 0.0039568087086081505
B: 1 V: 20 H: 420 W: 448
total_loss: 0.41953688859939575
train step 83; scene = ['dl3dv_7f5f6eac88ce53']; context = [[100, 122, 208, 167, 221, 234, 205, 184, 173, 131, 192, 237, 247, 230, 225, 139, 219, 136, 231, 250]]; loss = 0.419537;
scene scale: 1.452, pixel-wise num: 1806336, after voxelize: 1544955, voxelize ratio: 0.855
Gaussians attributes:
opacities: mean: 0.3320801556110382, min: 0.17441047728061676, max: 0.5498822927474976
scales: mean: 0.0014275726862251759, min: 0.0006626219837926328, max: 0.0032429553102701902
B: 1 V: 18 H: 224 W: 448
scene scale: 0.995, pixel-wise num: 1806336, after voxelize: 1138538, voxelize ratio: 0.630
Gaussians attributes:
opacities: mean: 0.36111998558044434, min: 0.18126343190670013, max: 0.5616821050643921
scales: mean: 0.001711647491902113, min: 0.0006804119911976159, max: 0.003537310753017664
B: 1 V: 16 H: 252 W: 448
total_loss: 0.5847789645195007
total_loss: 0.3895787000656128
train step 83; scene = ['dl3dv_1a8066e07d8dda']; context = [[276, 310, 279, 284, 286, 301, 303, 305, 306, 277, 294, 293, 285, 299, 304, 312]]; loss = 0.389579;
scene scale: 1.501, pixel-wise num: 1505280, after voxelize: 1406653, voxelize ratio: 0.934
Gaussians attributes:
opacities: mean: 0.2989327609539032, min: 0.0, max: 0.5765419006347656
scales: mean: 0.0012242008233442903, min: 0.0, max: 0.004018148873001337
B: 2 V: 10 H: 336 W: 448
total_loss: 0.26094043254852295
scene scale: 1.559, pixel-wise num: 2860032, after voxelize: 2758164, voxelize ratio: 0.964
Gaussians attributes:
opacities: mean: 0.26422399282455444, min: 0.13386768102645874, max: 0.5537470579147339
scales: mean: 0.0012501117307692766, min: 0.0006643979577347636, max: 0.0034388741478323936
B: 1 V: 24 H: 266 W: 448
total_loss: 0.7592687010765076
train step 84; scene = ['dl3dv_dfb6405ffe1652']; context = [[98, 275, 225, 208, 162, 228, 114, 266, 268, 181, 160, 147, 265, 140, 273, 248, 132, 155, 165, 128, 278, 127, 172, 299]]; loss = 0.759269;
scene scale: 1.148, pixel-wise num: 790272, after voxelize: 764951, voxelize ratio: 0.968
Gaussians attributes:
opacities: mean: 0.27898502349853516, min: 0.0, max: 0.5438324809074402
scales: mean: 0.0014081192202866077, min: 0.0, max: 0.003167996648699045
B: 4 V: 6 H: 294 W: 448

**Comment by yohan-pg** [comment]
Thank you. You are right, it does not always result in a batch size of 1. After some digging I found that `max_img_per_gpu: 24` can be used to control the batch size implicitly. (But I could not increase it without crashing with 80Gb of memory)
